### PR TITLE
Add hybrid IPs also

### PR DESCRIPTION
**Pull request by necessary129**
You people forgot to add hybrid IPs to the trusted list.
So just added it

### CODE REVIEW

**Comment by ProgVal** [comment]
Could you remove trailing whitespaces?

File: GitHub/plugin.py [hunk 1]
```diff
@@ -117,6 +117,11 @@ def doPost(self, handler, path, form):
                 not handler.address_string().startswith('192.30.252.') and \
                 not handler.address_string().startswith('192.30.253.') and \
                 not handler.address_string().startswith('192.30.254.') and \
+                not handler.address_string().startswith('::ffff:192.30.252.') and \ 
+                not handler.address_string().startswith('::ffff:192.30.253.') and \ 
+                not handler.address_string().startswith('::ffff:192.30.254.') and \ 
+                not handler.address_string().startswith('::ffff:192.30.255.') and \ 
+ 		        not handler.address_string().startswith('::ffff:204.232.175.') and \
```
Comment (t3chguy) [inline GitHub/plugin.py]: Something is wrong with your alignment on this line @necessary129

Comment (ProgVal) [inline GitHub/plugin.py]: they used tabulations

Comment (t3chguy) [inline GitHub/plugin.py]: mix of tabs and spaces in one file is bad enough, but thats in one line xD

**Comment by ProgVal** [comment]
Still some trailing spaces

**Comment by t3chguy** [comment]
There is still an alignment issue and 1 trailing space.

**Comment by necessary129** [comment]
Meh, I removed all that at the second time

**Comment by ProgVal** [comment]
There was still two tabulation characters, so I merged the PR manually: 1dd43c0.
I also changed the order of clauses.

**Comment by necessary129** [comment]
lol

**Comment by nyuszika7h** [comment]
Are you serious? This is not fixing the underlying issue, it's a hacky workaround which leads to unnecessary code duplication. `handler.address_string()` should strip `::ffff:` from the beginning of the IP before returning it.

**Comment by t3chguy** [comment]
:+1: Agree with @nyuszika7h

**Comment by ProgVal** [comment]
The best solution would be to implement  #230, anyway

**Comment by nyuszika7h** [comment]
But this can affect other plugins too, so we should handle IPv4-mapped addresses properly.

**Comment by ProgVal** [comment]
Indeed

**Comment by Mikaela** [comment]
>  The solution seems pretty obvious to me: handler.address_string() should strip ::ffff: from the beginning of the IP before returning it.

I disagree. The solution is fixing the HTTP server so it can bind both IPv4 and IPv6 at the same time. Do you remember seing this issue with any other server on your PC like SSH, Nginx, Apache, ZNC etc.?

**Comment by ProgVal** [comment]
I have a bot bound both to 127.0.0.1 and ::1. Can't you?

**Comment by Mikaela** [comment]
Comment updated. 127.0.0.1 and ::1 cannot be accessed externally (without reverse proxy in which case the authorized IPs of GitHub plugin would be useless as Limnoria doesn't understand the concept of reverse proxy (https://github.com/ProgVal/Limnoria/issues/894)).

**Comment by nyuszika7h** [comment]
> Do you remember seing this issue with any other server on your PC like SSH, Nginx, Apache, ZNC etc.?

No, I don't.

**Comment by ProgVal** [comment]
I just tried setting `supybot.servers.http.hosts4` to the empty string and `supybot.servers.http.hosts6` to `::`, and it binds all addresses. So what is the issue?

**Comment by Mikaela** [comment]
1. `config supybot.servers.http.hosts4 0.0.0.0`.
2. `config supybot.servers.http.hosts6 ::`.
3. Ensure that (sysctl) `net.ipv6.bindv6only` is `0` (the default).
4. Now start the bot and it should listen to both IPv4 and IPv6 without dotted-decimals or anything else.

And bonus question, where does your SSHd or web server or anything else than Limnoria's HTTPd listen on?

**Comment by Mikaela** [comment]
From my `/etc/ssh/sshd_config`:

```
ListenAddress ::
ListenAddress 0.0.0.0
```

From my nginx main server block:

```
    listen 80 default_server;
    listen [::]:80 default_server ipv6only=on;
    listen 443 default_server ssl;
    listen [::]:443 default_server ssl ipv6only=on;
```

From my any nginx vhost server block:

```
    listen 80;
    listen [::]:80;
    listen 443;
    listen [::]:443;
```

**Comment by ProgVal** [comment]
On 22/08/2015 12:21, Mikaela Suomalainen wrote:
> 1. `config supybot.servers.http.hosts4 0.0.0.0`.
> 2. `config supybot.servers.http.hosts6 ::`.
> 3. Ensure that (sysctl) `net.ipv6.bindv6only` is `0` (the default).

Don't you mean to set it to 1? http://serverfault.com/a/408670

> 4. Now start the bot and it should listen to both IPv4 and IPv6 without dotted-decimals or anything else.

If set to 0, it triggers an error (Address already in use), which is the
expected behavior

> And bonus question, where does your SSHd or web server or anything else than Limnoria's HTTPd listen on?

Pardon?

**Comment by Mikaela** [comment]
> Don't you mean to set it to 1? http://serverfault.com/a/408670

No, I mean to set it to 0 as Limnoria's HTTPd's buggy behaviour works with 1.

> If set to 0, it triggers an error (Address already in use), which is the expected behavior

>>  And bonus question, where does your SSHd or web server or anything else than Limnoria's HTTPd listen on?

Does this happen with your SSHd or non-Limnoria web server or any other working software that listens on IPv4 and IPv6 ports?

**Comment by Mikaela** [comment]
`socket.IPV6_V6ONLY` sounds like it's key to the proper fix, but I am still not a coder via \#IPv6.

**Comment by ProgVal** [comment]
On 22/08/2015 12:28, Mikaela Suomalainen wrote:
>> Don't you mean to set it to 1? http://serverfault.com/a/408670
> 
> No, I mean to set it to 0 as Limnoria's HTTPd's buggy behaviour works with 1.

When `net.ipv6.bindv6only` is `0` (ie. false), it means “don't bind only
ipv6”. So Limnoria does not bind only ipv6, ie. it binds ipv6 and ipv4.
So you have to keep supybot.servers.http.hosts4 empty because you can't
bind ipv4 addresses twice

When `net.ipv6.bindv6only` is `1` (ie. true), it means “bind only ipv6”,
so you can add addresses to supybot.servers.http.hosts4.

> 
>> If set to 0, it triggers an error (Address already in use), which is the expected behavior
> 
>>>  And bonus question, where does your SSHd or web server or anything else than Limnoria's HTTPd listen on?
> 
> Does this happen with your SSHd or non-Limnoria web server or any other working software that listens on IPv4 and IPv6 ports?

Because they **override** `net.ipv6.bindv6only` by using IPV6_V6ONLY

**Comment by Mikaela** [comment]
> Because they **override** `net.ipv6.bindv6only` by using IPV6_V6ONLY

And this is exactly what Limnoria should do.

**Comment by ProgVal** [comment]
I don't think a software should override a system default without the user explicitely asking for it

**Comment by Mikaela** [comment]
> I don't think a software should override a system default without the user explicitely asking for it

Not doing it causes misbehaviour such as the dotted decimals and can cause confusion. I believe SSH and Nginx & co are right in what they do and they haven't asked my permission for it either.

**Comment by ProgVal** [comment]
Then you should ask your OS to set `net.ipv6.bindv6only` to 1 as a default.

**Comment by Mikaela** [comment]
> Then you should ask your OS to set net.ipv6.bindv6only to 1 as a default.

This is what I have been doing for months, but this only fixes Limnoria's misbehaviour for me, not other users.

I have no further comment on this issue, whether you care about users or not is your choice.

**Comment by nyuszika7h** [comment]
What's the problem now? The handling of IPv4-mapped addresses has been fixed.

**Comment by nyuszika7h** [comment]
I found something interesting: http://security.stackexchange.com/questions/92081/is-receiving-ipv4-connections-on-af-inet6-sockets-insecure

**Comment by ProgVal** [comment]
I'll add a config variable for that.
Fine?

**Comment by Mikaela** [comment]
> What's the problem now? The handling of IPv4-mapped addresses has been fixed.

The problem is that as user who has `net.ipv6.bindv6only` set to `1` Limnoria defaults to having HTTPd listen only on IPv6 which can be a little confusing. Also what if the default value of that changes in future kernel or changes to 1 and is removed entirely?

> I'll add a config variable for that.
> Fine?

Yes, **if** Limnoria starts listening on `0.0.0.0` **and** `::` **and** works regardless of what the system administrator has set `net.ipv6.bindv6only` to.

In case there are nothing more where I should reply, I don't comment further like I said previously.

**Comment by ProgVal** [comment]
good point

**Comment by nyuszika7h** [comment]
I think because of the possible security issue mentioned above, we should listen on `0.0.0.0`, and `::` with ipv6only. That should work everywhere too, so I don't necessarily think a config variable is needed.

**Comment by ProgVal** [comment]
Done, with defaults recommended by @Mikaela: https://github.com/ProgVal/Limnoria/commit/a39238aba9489ef69fa4f7d545502a4049ac1953

@nyuszika7h: Being configurable doesn't hurt. It may also be slightly faster to have a single dual-stack server than two single-stack ones.

**Pull request closed without merging**